response_operations_ui/controllers/message_controllers.py

```diff
@@ -61,12 +59,23 @@ def get_conversation_count(params):
         raise NoMessagesError
 
 
-def get_thread_list(params):
-    logger.info("Retrieving threads list")
+def _get_secure_message_threads_params(survey_id, business_id, conversation_tab):
+    """creates a params dictionary"""
+    params = {'survey': survey_id,
+              'is_closed': 'true' if conversation_tab == 'closed' else 'false',
+              'my_conversations': 'true' if conversation_tab == 'my messages' else 'false',
+              'new_respondent_conversations': 'true' if conversation_tab == 'initial' else 'false'}
+    if business_id:
+        params['business_id'] = business_id
+    return params
+
 
-    params['is_closed'] = 'true' if params.get('is_closed') else 'false'
-    params['my_conversations'] = 'true' if params.get('my_conversations') else 'false'
-    params['new_respondent_conversations'] = 'true' if params.get('new_respondent_conversations') else 'false'
+def get_thread_list(survey_id, business_id, conversation_tab, page, limit):
+
+    logger.info("Retrieving threads list")
```

Comment:
More metadata here would be useful.  Any/all of survey_id, business_id and conversation_tab would all aid in debugging

Reply:
Added

Reply:
Added